src/main/java/io/tiledb/java/api/QueryCondition.java

```diff
@@ -84,6 +86,58 @@ public QueryCondition(
     this.ctx = ctx;
   }
 
+  /**
+   * Constructor
+   *
+   * @param ctx The context
+   * @param attributeName The name of the field this operation applies to
+   * @param type The datatype
+   * @param value The value to compare to
+   * @param OP The relational operation between the value of the field and `condition_value`
+   * @throws TileDBError
+   */
+  public QueryCondition(
+      Context ctx,
+      String attributeName,
+      Datatype type,
+      Object value,
+      tiledb_query_condition_op_t OP)
+      throws TileDBError {
+    try {
+      conditionpp = tiledb.new_tiledb_query_condition_tpp();
+      ctx.handleError(tiledb.tiledb_query_condition_alloc(ctx.getCtxp(), conditionpp));
+      NativeArray array = null;
+      if (value.getClass().isArray()) {
+        array = new NativeArray(ctx, value, type.javaClass());
+      } else {
+        if (type != null) {
+          int byteSize = type.getNativeSize();
+          array = new NativeArray(ctx, byteSize, type);
+          array.setItem(0, value);
+        }
+      }
+      conditionp = tiledb.tiledb_query_condition_tpp_value(conditionpp);
+      if (type == null) {
```

Comment:
The logic is a little bit hard to follow here. It seems like you could simplify quite a bit by moving this check to the top?

Reply:
I have simplified the new constructor